[CS2113-T13-4] SuperTracker

rismm opened the pull request:
SuperTracker helps to streamline the process of checking inventory for supermarket inventory managers.

BestDownLoader365 commented on docs/uml-diagrams/FileManager.puml:
```diff
@@ -0,0 +1,47 @@
+    +Item(name:String, quantity:int,
+    price:double, expiryDate:LocalDate)
+    +getName():String
+}
```
Should you explain why `SuperTracker` depend on `FileManager`?

BestDownLoader365 commented on docs/uml-diagrams/FileManager.puml:
```diff
@@ -0,0 +1,47 @@
+Item <. FileManager
+Item "*" <-* Inventory
+FileManager <.. Command
+Inventory <.. Command
```
Should you use standard docorator to represent the class type such as `interface` or basic `class`.

BestDownLoader365 commented on docs/uml-diagrams/ListCommand.puml:
```diff
@@ -0,0 +1,145 @@
+Ui --> ListCommand
+deactivate Ui
+
+end
```
In all your sequence diagram, only objects are interacting. So should you remove the <<`class`>> decorator above?

BestDownLoader365 commented on docs/uml-diagrams/ListCommand.puml:
```diff
@@ -0,0 +1,145 @@
+loop items
+
+ListCommand -> Ui : listItem(item:Item, index:int, hasQuantity:boolean, hasPrice:boolean, hasExpiry:boolean, firstParam:String, secondParam:String):void)
+activate Ui #e5c2ea
```
For the sequnce diagram, its too big, can you split it into small parts?

edwardhumi commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,251 @@
+- `FileManager`: To save the new item added onto the hard disk
+
+The following sequence diagram shows the execution of a NewCommand<br>
+![NewCommandSequence](uml-diagrams/NewCommandSequence.png)
```
For the `new Item(...)`, according to the module website, a constructor does not need a `new` keyword

edwardhumi commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,251 @@
+5. `sortBy == "p"` and `isReverse == false`: Price ascending
+6. `sortBy == "p"` and `isReverse == true`: Price descending
+
+The following sequence diagram shows the execution of a ListCommand<br>  
+![ListCommandSequence](uml-diagrams/ListCommandSequence.png)
```
After calling a constructor, `new ArrayList<>...`, you can mark X in the end of the diagram to show that it is deleted

wallywallywally commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+Converting the parameter string to a pattern is done because **order does not matter** in the user's input parameters _(if there are multiple parameters)._
+Thus, it would be necessary to convert the user's input into a consistent format for pattern matching using regex.
+
+> Note: 
+> - All command regex are in the format:\
+>   `a/(?<grp1>.*) b/(?<grp2>.*) c/(?<grp3>.*) ` with `a/`, `b/`, `c/` being the respective parameter flags
+>   and `/` as the flag character.
+>   - As of now, parameter flags must only contain 1 character right before the `/`
+>   - Suppose we have `a/aaa b/bbb c/ccc ` as an input, according to the regex above, 
+>   aaa will be in the named capture group "grp1", bbb in "grp2", ccc in "grp3"
+> - If a parameter is optional, we would have it in a non-capture group with a `?` at the end
+>   - i.e. in `a/(?<grp1>.*) (?<grp2>(?:b/.*)?)`, the `b/` flag is an optional parameter
+> 
+> For example, suppose the program wants to match the user input parameter
+> `c/coconut a/apple b/bear a/anaconda d/donkey` to the regex `a/(?<grp1>.*) b/(?<grp2>.*) c/(?<grp3>.*) (?<grp4>(?:e/.*)?)`
+> 1. The input parameter string and a string array of flags `{a, b, c, e}` is passed into `Parser-makeStringPattern(...)`
+> 2. `Parser-makeStringPattern` will return a pattern string `a/apple b/bear c/coconut `
+> 3. The program will then try to match the pattern string `a/apple b/bear c/coconut ` to the regex `
+> a/(?<grp1>.*) b/(?<grp2>.*) c/(?<grp3>.*) (?<grp4>(?:e/.*)?)`
+> 4. The above pattern string matches the regex, with the following strings in the named capture groups:
+>    - "grp1": apple
+>    - "grp2": bear
+>    - "grp3": coconut
+>    - "grp4" is empty
+
```
The explanation here feels repetitive, and can be shortened.

wallywallywally commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+
+This string is generated by the method `FileManager-getItemData()`
+
+#### Loading
+Loading is performed at the start of the program in `SuperTracker-run()` where it calls the method `FileManager+loadData()`.
+`loadData()` looks for the save text file at `./data/items.txt` and reads the lines from the file `items.txt` exists. As indicated earlier,
+each line containing a single `Item`'s data will be in the format and order of `NAME,QUANTITY,PRICE,EXPIRY_DATE`. Each attribute will be parsed into its
+relevant data type, creating a new `Item` with the extracted attributes, which is then added to the item list in `Inventory`. In the event where the line
+of data read is not in the correct format, or the attributes are unable to be parsed into its relevant data type (i.e. the string in the QUANTITY part reads "f1ve" instead of "5")_,
+the line will be ignored and a new `Item` will not be added to the list.
+
+The parsing of string data to an `Item` object is handled by the method `FileManager-parseItemData()`.
```
This explanantion can be formatted in a way that is easier to read, perhaps using a list.

wallywallywally commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+Saving and loading data of `Item` objects is performed by the `FileManager` class. As the program is running, 
+`Item` objects will be stored in the `Inventory` class. The following is a class diagram of `FileManager`
+and its relevant dependencies\
+![FileManagerClass](./uml-diagrams/FileManagerClass.png)
```
Class diagram is too complicated, can remove redundant attributes or methods to make it easier to understand for this implementation.

wallywallywally commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+
+### New Command
+The following is a class diagram of the NewCommand and its relevant dependencies<br>
+![NewCommandClass](uml-diagrams/NewCommandClass.png)
```
Sequence diagram is too complicated. The `new Item(...)` call is too long and can be simplified.

wallywallywally commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+- `Ui`: To notify the user about the successful execution of `DeleteCommand`
+
+The following sequence diagram shows the execution of a DeleteCommand<br>
+![DeleteCommandSequence](uml-diagrams/DeleteCommandSequence.png)
```
Activation bar for `Inventory` is not consistent. `contains()` is green while `delete()` is white.

wallywallywally commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+The following sequence diagram shows the execution of a NewCommand<br>
+![NewCommandSequence](uml-diagrams/NewCommandSequence.png)
+
+1. The `SuperTracker` class calls the `execute` method of `NewCommand`
```
`SuperTracker` is not explained enough previously to let the reader understand it is responsible for a lot of the main functionality.

wallywallywally commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+
+### List Command
+The following is a class diagram of the FindCommand and its relevant dependencies<br>
+![ListCommandClass](uml-diagrams/ListCommandClass.png)
```
Class diagram is too complicated. I think it can be broken down into smaller parts in other diagrams and displayed when those parts are being explained.

wallywallywally commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+- `Inventory`: For getting the list of items in the inventory
+- `Ui`: To print the list of items in the inventory to the output
+
+The 7 parameters in the constructor `hasQuantity`, `hasPrice`, `hasExpiry`, `firstParam`, `secondParam`, `sortBy`, `isReverse` are used to determine how the list should be printed out.
```
Explanation is too long and can be presented in a more readable way, perhaps a table.

wallywallywally commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+- `isReverse`
+  - True if the user has input the reverse flag `r/`, false otherwise.
+
+There are 8 sorting modes in total
+1. `sortBy == ""` and `isReverse == false`: Alphabetical ascending (e.g. A-Z)
+2. `sortBy == ""` and `isReverse == true`: Alphabetical descending (e.g. Z-A)
+3. `sortBy == "q"` and `isReverse == false`: Quantity ascending
+4. `sortBy == "q"` and `isReverse == true`: Quantity descending
+5. `sortBy == "p"` and `isReverse == false`: Price ascending
+6. `sortBy == "p"` and `isReverse == true`: Price descending
+7. `sortBy == "e"` and `isReverse == false`: Expiry date ascending (e.g. 2024-2025)
+8. `sortBy == "e"` and `isReverse == true`: Expiry date descending (e.g. 2025-2024)
```
Explanation is not very easy, can use a table perhaps.

wallywallywally commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+8. `sortBy == "e"` and `isReverse == true`: Expiry date descending (e.g. 2025-2024)
+
+The following sequence diagram shows the execution of a ListCommand<br>  
+![ListCommandSequence](uml-diagrams/ListCommandSequence.png)
```
This sequence diagram is way too large and complicated.

wallywallywally reviewed (commented):
Diagrams and explanations can be made more readable, but I like how thorough the design and implementation is!

hwc0419 commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Input Parsing
+The program handles user inputs in the Parser class where inputs are parsed into command classes that implement the Command interface. 
+The Parser class only contains static methods as we have determined that it would be more practical instead of instantiating an instance of a Parser class.
```
Do you mind explaining why it would be more practical?

hwc0419 commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+Take note that the `Command` class in the diagram refers to the classes that implement the `Command` interface and make changes to the item list in `Inventory`
+
+#### Saving
+Saving is performed automatically each time the item list is updated. Currently, there are only 5 commands that can make changes to
+the item list and will call `FileManager+saveData()` at the end of `Command+execute()`.
```
Will the entire save file be overwritten each time one of the commands are ran, or only the part that is being changed?

hwc0419 commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+
+#### Loading
+Loading is performed at the start of the program in `SuperTracker-run()` where it calls the method `FileManager+loadData()`.
+`loadData()` looks for the save text file at `./data/items.txt` and reads the lines from the file `items.txt` exists. As indicated earlier,
```
What happens if the directory 'data' or the file 'items.txt' doesn't exist?

hwc0419 commented on docs/DeveloperGuide.md:
```diff
@@ -5,9 +5,262 @@
+> - All command regex are in the format:\
+>   `a/(?<grp1>.*) b/(?<grp2>.*) c/(?<grp3>.*) ` with `a/`, `b/`, `c/` being the respective parameter flags
+>   and `/` as the flag character.
+>   - As of now, parameter flags must only contain 1 character right before the `/`
```
Maybe it would be good to handle the error if the user enters more than 1 character, or edit the regex pattern string to specify that there must be only one character

dtaywd committed:
Merge remote-tracking branch 'upstream/master' into expenditure-command-incorrect-error-handling

rismm committed:
Merge pull request #222 from dtaywd/expenditure-command-incorrect-error-handling

Expenditure command incorrect error handling

Commit:
Authortags

rismm committed:
Merge pull request #223 from TimothyLKM/master

Author tags

rismm committed:
Add PPP

Edit UG for file saving

awesomesjh committed:
Add clear command

awesomesjh committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-4/tp into add-clear-command

# Conflicts:
#	src/main/java/supertracker/parser/Parser.java

awesomesjh committed:
Rearrange variables and fix checkstyle

awesomesjh committed:
Add tests for clear command

rismm committed:
Update UG

awesomesjh committed:
Add clear command to user guide

awesomesjh committed:
Add clear command to help

awesomesjh committed:
Merge pull request #225 from rismm/documentation

Update docs

dtaywd committed:
Merge pull request #226 from awesomesjh/add-clear-command

Add clear command

dtaywd committed:
add PPP

dtaywd committed:
add more differentiated ways of handling error for revenue and expenditure

vimalapugazhan committed:
Merge pull request #230 from dtaywd/PPP

Ppp

rismm committed:
Merge branch 'master' into transaction-saving

rismm committed:
Add file saving after clear command

vimalapugazhan committed:
Add ProfitCommand

vimalapugazhan committed:
Merge pull request #231 from rismm/transaction-saving

Add saving after clear

vimalapugazhan committed:
Merge branch 'master' into Vimal-ProfitCommand

dtaywd committed:
update ppp with correct link

vimalapugazhan committed:
Fix checkstyle

TimothyLKM committed:
Merge pull request #232 from vimalapugazhan/Vimal-ProfitCommand

Vimal profit command

dtaywd committed:
add JUnit Test for expenditure and revenue command

dtaywd committed:
add Javadocs for JUnit tests

dtaywd committed:
add javadocs

dtaywd committed:
update test value

dtaywd committed:
update test value

awesomesjh committed:
Add PPP

vimalapugazhan committed:
Merge pull request #234 from awesomesjh/add-ppp

Add PPP

dtaywd committed:
update apple to grape

dtaywd committed:
Merge remote-tracking branch 'upstream/master' into JUnit-Exp-rev

# Conflicts:
#	docs/team/dtaywd.md

dtaywd committed:
Add transactions clear

awesomesjh committed:
Update src/main/java/supertracker/command/ReportCommand.java

awesomesjh committed:
Merge pull request #233 from dtaywd/JUnit-Exp-rev

J unit exp rev

awesomesjh committed:
Update add and remove commands to throw error when quantity is 0

vimalapugazhan committed:
Add JavaDocs for RevenueCommand

awesomesjh committed:
Update list command

Sorting by quantity/price/expiry now displays quantity/price/expiry as well

vimalapugazhan committed:
Add JavaDocs for DeleteCommand Class

vimalapugazhan committed:
Add JavaDocs for ProfitCommand Class

vimalapugazhan committed:
Add JavaDocs for printProfit, printRevenueExpenditure, deleteCommandSuccess and getExpiryDateMessage methods in Ui class.

awesomesjh committed:
Temp commit

vimalapugazhan committed:
Add JavaDocs to Revenue Profit and related methods and parseDate, parseExpiryDateUpdate

vimalapugazhan committed:
Add profit command into UG

dtaywd committed:
Merge pull request #235 from vimalapugazhan/Vimal-Java-Docs,UG

Vimal JavaDocs and UG for Profit Command

awesomesjh committed:
Fix bug in list command

awesomesjh committed:
Add more list command tests

awesomesjh committed:
Rearrange imports

awesomesjh committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-4/tp into fix-list-command

# Conflicts:
#	src/main/java/supertracker/parser/Parser.java

awesomesjh committed:
Fix checkstyle

rismm committed:
Merge pull request #236 from awesomesjh/fix-list-command

Fix list command

vimalapugazhan committed:
Start PPP

vimalapugazhan committed:
Merge branch 'master' into Vimal-PPP

vimalapugazhan committed:
Format PPP

Commit:
Timothy-PPP

Commit:
Merge branch 'master' into Timothy-PPP

vimalapugazhan committed:
Fix Format of PPP

awesomesjh committed:
Refactor exp, rev, profit commands in Parser

vimalapugazhan committed:
Merge pull request #237 from TimothyLKM/Timothy-PPP

Timothy's ppp

vimalapugazhan committed:
Merge pull request #238 from awesomesjh/refactor-parser

Refactor exp, rev, profit commands in Parser

vimalapugazhan committed:
Merge branch 'master' into Vimal-PPP

awesomesjh committed:
Update user guide table

awesomesjh committed:
Change profit to local variable

vimalapugazhan committed:
no message

awesomesjh committed:
Update about us

awesomesjh committed:
Add profit command to help

awesomesjh committed:
Add Javadocs

Commit:
Rename Timothy's PPP to lower case
Add Javadocs for HelpCommand and HelpCommandUi

Commit:
create JUnit for Rename Command

vimalapugazhan committed:
no message

rismm committed:
Merge pull request #242 from vimalapugazhan/Vimal-PPP

awesomesjh committed:
Remove closing </p> tags in Javadocs

awesomesjh committed:
Merge branch 'master' into add-javadoc

Commit:
Fix bug in rename

Commit:
Add more JUnit tests for rename command

Commit:
Merge branch 'master' into Javadocs

Commit:
Change method name

TimothyLKM committed:
Merge pull request #240 from awesomesjh/add-javadoc

Add Javadocs

Commit:
Merge branch 'master' into Javadocs

# Conflicts:
#	src/main/java/supertracker/parser/Parser.java
#	src/main/java/supertracker/ui/HelpCommandUi.java

vimalapugazhan committed:
Merge pull request #243 from TimothyLKM/Javadocs

Add Javadocs and various edits

rismm committed:
Update DG with user stories

TimothyLKM committed:
Merge pull request #245 from rismm/documentation

Update DG with user stories

awesomesjh committed:
Fix rename command if new name already exists in inventory

TimothyLKM committed:
Merge pull request #246 from awesomesjh/fix-rename-command

Fix rename command if new name already exists in inventory

Commit:
add UG

Commit:
Merge branch 'master' into update-ug

awesomesjh committed:
Add JUnit tests for buy and sell commands

rismm committed:
Merge pull request #247 from awesomesjh/add-buy-sell-test

Add JUnit tests for buy and sell commands

Commit:
various edits

awesomesjh committed:
Replace magic numbers

rismm committed:
Merge pull request #248 from TimothyLKM/update-ug

Javadocs and some edits to the code.

awesomesjh committed:
Fix checkstyle

awesomesjh committed:
Merge pull request #249 from awesomesjh/refactor-parser

Replace magic numbers

rismm committed:
Add JUnit for transaction data loading

awesomesjh committed:
Add valid expiry date range to user guide

vimalapugazhan committed:
Merge pull request #250 from rismm/master

Add JUnit for transaction data loading

vimalapugazhan committed:
Merge pull request #251 from awesomesjh/modify-ug

Add valid expiry date range to user guide

Commit:
Finish my UG

Commit:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-4/tp

TimothyLKM committed:
Merge pull request #252 from TimothyLKM/master

Finished UG

Commit:
Finished all my UML diagrams

vimalapugazhan committed:
Add JUnit for profit command and edit constants in revenue and expediture command junits

vimalapugazhan committed:
Merge branch 'master' into JUnit-ProfitCommand

vimalapugazhan committed:
Fix JavaDocs RevenueJUnit

vimalapugazhan committed:
fix checkstyle

rismm committed:
Merge pull request #253 from vimalapugazhan/JUnit-ProfitCommand

J unit ProfitCommand

rismm committed:
Add javadocs to storage classes

rismm committed:
Update TransactionStorage.java

Co-authored-by: vimalapugazhan <111181615+vimalapugazhan@users.noreply.github.com>

vimalapugazhan committed:
Merge pull request #254 from rismm/master

Add javadocs to storage classes

awesomesjh committed:
Fix issue #257

vimalapugazhan committed:
Merge pull request #261 from awesomesjh/modify-ug

Fix issue #257

awesomesjh committed:
Add more details about command format

awesomesjh committed:
Change user guide description

awesomesjh committed:
Update TOC

awesomesjh committed:
Update README

Commit:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-4/tp

rismm committed:
Merge pull request #262 from awesomesjh/modify-ug

Add more details about command format

rismm committed:
Fix file saving bugs

rismm committed:
Update TransactionStorage.java

rismm committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-4/tp

Commit:
Finish DG

Commit:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-4/tp

Commit:
remove blank line

rismm committed:
Update UG

rismm committed:
Update JUnit tests

rismm committed:
Update DG

Update DG to reflect changes to file saving

rismm committed:
Merge pull request #263 from TimothyLKM/master

Finish DG for rename and help command

rismm committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-4/tp

rismm committed:
Update DG

Better english

rismm committed:
Update again

Small fix

rismm committed:
Update again

Another small fix

vimalapugazhan committed:
Merge pull request #264 from rismm/master

Fix file saving bugs

vimalapugazhan committed:
Update PPP and fix capitalized variable forJUnit test

vimalapugazhan committed:
Merge branch 'master' into Update-PPP

awesomesjh committed:
Update remove and sell commands

- A different message is now displayed if the quantity removed or sold is 0
- If the quantity sold is 0, no transaction would be added to the transaction list

rismm committed:
Merge pull request #265 from awesomesjh/update-remove-command

Update remove and sell commands

dtaywd committed:
fix typo in exp, rev and profit test

dtaywd committed:
update ppp

dtaywd committed:
add reportCommand to the developer guide

vimalapugazhan committed:
Merge branch 'master' into Update-PPP

awesomesjh committed:
Fix typo

awesomesjh committed:
Add acknowledgements

dtaywd committed:
add expenditure command to developer guide

dtaywd committed:
update ppp

dtaywd committed:
Merge pull request #266 from awesomesjh/modify-dg

Update developer guide

dtaywd committed:
Merge remote-tracking branch 'upstream/master' into dev-guide-update

vimalapugazhan committed:
Merge pull request #267 from dtaywd/dev-guide-update

Dev guide update

vimalapugazhan committed:
DG update for RevenueCommand

vimalapugazhan committed:
Merge branch 'master' into Update-PPP

# Conflicts:
#	docs/DeveloperGuide.md

rismm committed:
Add architecture diagram

vimalapugazhan committed:
Add Revenue and Profit UML, Edit return type in other Command UMLs

vimalapugazhan committed:
edit profit command in DG

vimalapugazhan committed:
Edit PPP

vimalapugazhan committed:
Revert Changes

awesomesjh committed:
Merge pull request #268 from vimalapugazhan/Update-PPP_DG

Update PPP, DG and UML

dtaywd committed:
update update command uml

rismm committed:
Merge branch 'master' into documentation

dtaywd committed:
update parameters in update uml

vimalapugazhan committed:
Merge pull request #269 from dtaywd/update-update-command-uml

update update command uml

rismm committed:
Add appendix in DG

rismm committed:
Add architecture diagram to DG

rismm committed:
Add more details to UG

We are not responsible for data loss if the user deletes the file while program is running

vimalapugazhan committed:
Update DG and UML for Delete.

vimalapugazhan committed:
Merge pull request #270 from rismm/documentation

Update DG

rismm committed:
Merge pull request #271 from vimalapugazhan/Update-DG2

Update DG and UML for Delete.

awesomesjh committed:
Refactor clear command

awesomesjh committed:
Add UML diagrams

awesomesjh committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-4/tp into modify-dg

awesomesjh committed:
Add clear command to developer guide

awesomesjh committed:
Update sell command and developer guide

awesomesjh committed:
Add TOC to developer guide

vimalapugazhan committed:
Merge pull request #272 from awesomesjh/modify-dg

Update developer guide

awesomesjh committed:
Update docs

awesomesjh committed:
Fix bugs

awesomesjh committed:
Fix UML diagrams

awesomesjh committed:
Update section headers

awesomesjh committed:
Update appendix

vimalapugazhan committed:
Merge pull request #273 from awesomesjh/modify-dg

Fix UML diagrams and developer guide

dtaywd committed:
update report command uml

dtaywd committed:
update dev guide

dtaywd committed:
update report sequence diagram

awesomesjh committed:
Add instructions for manual testing

awesomesjh committed:
Merge pull request #274 from dtaywd/uml-update

Uml update

dtaywd committed:
Merge pull request #275 from awesomesjh/modify-dg

Add instructions for manual testing

awesomesjh committed:
Add more instructions for manual testing

awesomesjh committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-4/tp into modify-dg

dtaywd committed:
Merge pull request #276 from awesomesjh/modify-dg

Add more instructions for manual testing

dtaywd committed:
change startDate and endDate to START_DATE and END_DATE

dtaywd committed:
fix nd to end

awesomesjh committed:
Merge pull request #277 from dtaywd/ug-minor-update

change startDate and endDate to START_DATE and END_DATE

awesomesjh committed:
Fix author tags

awesomesjh committed:
Fix author tags

awesomesjh committed:
Replace empty strings with constant

awesomesjh committed:
Refactor extractParam

awesomesjh committed:
Refactor code

awesomesjh committed:
Fix sequence diageam for rename command

awesomesjh committed:
Fix sequence diagram for profit command

awesomesjh committed:
Update description for add and remove commands

awesomesjh committed:
Fix bug in remove command description

awesomesjh committed:
Update PPP

awesomesjh committed:
Fix UML diagrams

awesomesjh committed:
Merge pull request #278 from awesomesjh/fix-author-tags

Fix author tags

awesomesjh committed:
Fix UML diagrams

awesomesjh committed:
Merge pull request #279 from awesomesjh/modify-dg

Fix UML diagrams

awesomesjh committed:
Update PPP

awesomesjh committed:
Merge pull request #280 from awesomesjh/modify-dg

Update PPP

rismm committed:
Update PPP

rismm committed:
Merge pull request #281 from rismm/master

Update PPP

awesomesjh committed:
Fix list command

awesomesjh committed:
Merge pull request #282 from awesomesjh/modify-dg

Fix list command

awesomesjh committed:
Remove spacing after <br> tag

awesomesjh committed:
Fix clear command

awesomesjh committed:
Merge pull request #283 from awesomesjh/modify-dg

Remove spacing after <br> tag

Commit:
Update PPP

rismm committed:
Update rename command

awesomesjh committed:
Merge pull request #284 from TimothyLKM/master

Update PPP

awesomesjh committed:
Merge pull request #285 from rismm/master

Update rename command

awesomesjh committed:
Update rename

rismm committed:
Merge pull request #286 from awesomesjh/update-rename

Update rename

awesomesjh committed:
Fix report output

awesomesjh committed:
Update report test in developer guide

awesomesjh committed:
Remove magic literal

awesomesjh committed:
Fix report command test

rismm committed:
Merge pull request #287 from awesomesjh/update-rename

Update report output

awesomesjh committed:
Change example for clear command

dtaywd committed:
Merge pull request #288 from awesomesjh/update-rename

Change example for clear command

dtaywd committed:
add line break

awesomesjh committed:
Fix class diagrams for exp and rev commands

dtaywd committed:
Merge pull request #290 from awesomesjh/update-rename

Fix class diagrams for exp and rev commands

awesomesjh committed:
Merge pull request #289 from dtaywd/dev-guide-line-break

add line break

dtaywd committed:
update dev guide by removing line break

dtaywd committed:
Merge branch 'dev-guide-line-break' into dev-guide-update1

awesomesjh committed:
Merge pull request #291 from dtaywd/dev-guide-update1

Dev guide update1

awesomesjh committed:
Fix delete command class diagram

rismm committed:
Merge pull request #292 from awesomesjh/update-rename

Fix delete command class diagram

awesomesjh committed:
Update report command sequence diagram

awesomesjh committed:
Fix report command sequence diagram

awesomesjh committed:
Merge pull request #293 from awesomesjh/update-rename

Update report command sequence diagram

awesomesjh committed:
Fix headers

awesomesjh committed:
Merge pull request #294 from awesomesjh/update-rename

Fix headers

awesomesjh committed:
Add missing commands

awesomesjh committed:
Rename file paths

rismm committed:
Merge pull request #295 from awesomesjh/update-rename

Add missing commands

awesomesjh committed:
Fix file name

awesomesjh committed:
Fix test case

rismm committed:
Merge pull request #296 from awesomesjh/update-rename

Fix file name

awesomesjh committed:
Realign example

awesomesjh committed:
Merge pull request #297 from awesomesjh/update-rename

Realign example
